test/jdk/javax/net/ssl/templates/TLSBase.java

```diff
@@ -67,42 +64,77 @@ abstract public class TLSBase {
 
     TLSBase() {
         String keyFilename =
-                System.getProperty("test.src", "./") + "/" + pathToStores +
-                        "/" + keyStoreFile;
+            System.getProperty("test.src", "./") + "/" + pathToStores +
+                "/" + keyStoreFile;
         String trustFilename =
                 System.getProperty("test.src", "./") + "/" + pathToStores +
-                        "/" + trustStoreFile;
+                    "/" + trustStoreFile;
         System.setProperty("javax.net.ssl.keyStore", keyFilename);
         System.setProperty("javax.net.ssl.keyStorePassword", passwd);
         System.setProperty("javax.net.ssl.trustStore", trustFilename);
         System.setProperty("javax.net.ssl.trustStorePassword", passwd);
     }
 
     // Base read operation
-    byte[] read(SSLSocket sock) {
-        try {
-            BufferedReader reader = new BufferedReader(
-                    new InputStreamReader(sock.getInputStream()));
-            String s = reader.readLine();
-            System.err.println("(read) " + name + ": " + s);
-            return s.getBytes();
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-        return null;
+    byte[] read(SSLSocket sock) throws Exception {
+        BufferedReader reader = new BufferedReader(
+            new InputStreamReader(sock.getInputStream()));
+        String s = reader.readLine();
+        System.err.println("(read) " + name + ": " + s);
+        return s.getBytes();
     }
 
     // Base write operation
-    public void write(SSLSocket sock, byte[] data) {
-        try {
-            PrintWriter out = new PrintWriter(
-                    new OutputStreamWriter(sock.getOutputStream()));
-            out.println(new String(data));
-            out.flush();
-            System.err.println("(write)" + name + ": " + new String(data));
-        } catch (Exception e) {
-            e.printStackTrace();
+    public void write(SSLSocket sock, byte[] data) throws Exception {
+        PrintWriter out = new PrintWriter(
+            new OutputStreamWriter(sock.getOutputStream()));
+        out.println(new String(data));
+        out.flush();
+        System.err.println("(write)" + name + ": " + new String(data));
+    }
+
+    private static KeyManager[] getKeyManager(boolean empty) throws Exception {
+        FileInputStream fis = null;
+        if (!empty) {
+            fis = new FileInputStream(System.getProperty("test.src", "./") + "/" + pathToStores +
+                "/" + keyStoreFile);
         }
+        // Load the keystore
+        char[] pwd = passwd.toCharArray();
+        KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
+        ks.load(fis, pwd);
```

Comment:
Should this input stream be closed?

Reply:
`fis` can be null, so I'd need a check.  Since this is just a quick test, it is not using many resources and decided to let exiting of the test close it.